src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,214 @@
+package duke.parser;
+
+import duke.exceptions.InvalidTypeException;
+import duke.exceptions.MissingNameException;
+import duke.exceptions.MissingEventDateException;
+import duke.exceptions.MissingDeadlineDateException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.ui.Ui;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    private String userInput;
+
+    public Parser(String userInput) {
+        this.userInput = userInput;
+    }
+
+    /**
+     * Process user commands.
+     *
+     * @return False if the command is "bye", True otherwise
+     */
+    public boolean processInput() {
```

Comment:
Should this return a boolean? Based on my understanding, I think methods that return a boolean should also follow the `isXYZ()` naming convention (eg. `isExit()`, `isDone()`).

Reply:
Perhaps you could consider separating this into 2 functions where 1 solely handles displaying the correct response while another handles checking whether the application should exit?